### ISSUE TITLE
[RFC]  Move to use main as default branch 

### ISSUE THREAD

**Issue by tqchen** [dmlc/xgboost#6211]
In light of recent proposed change of [github's default branch from master to main](https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/). It would be great for the community to consider such a change. Given the minimum impact to set it up. We believe is a net positive to the community and also keeps us up to the latest github convention.

This thread seeks formal lazy consensus :) Everyone is also more than welcomed to share their thoughts.

I experimented the change with several repos, and the steps are pretty easy:

- create main branch from master
- change branch protection rule from master to main
- New PRs and other activities will continue to work as normal
- For stale PRs that still need updates, reset the base to main 
   - In github UI, click edit, branches, change from master to main

**Comment by RAMitchell** [comment]
Rapids moved to main already and it was fairly painless.

**Comment by hcho3** [comment]
I suggest that we pick a time that's not close to a release date. The CI needs to be updated to recognize the new main branch and there might be some hiccups along the way.

**Comment by tqchen** [comment]
@hcho3 For Jenkins CI no changes should be needed. I am not too sure about other CI though. happy to wait until the current release is out

**Comment by hcho3** [comment]
@tqchen It has more to do with Continuous Delivery (CD) pipeline that generates artifacts for the master branch.

**Comment by trivialfis** [comment]
I don't think we are going anywhere at the moment. Feel free to reopen if this is still desired.
